### DISCUSSION TITLE
Node-RED and Home Assistant running in different namespaces

### DISCUSSION THREAD

**Discussion by rjdza** [zachowj/hass-node-red#90] [Q&A]
I have Home Assistant running under docker (with the HACS Node-RED companion installed).  

Node-RED is running directly on the host, with the contrib nodes installed.  

Is this a supported configuration?  If so, is there a reason it doesn't see the hass client?  

If not, is there a workaround? How does the components communicate?

Finally, not really related, occasionally (frequently) when I deploy, one or two nodes say they cannot communicate with the server.  Is there a solution to this, or will I have to downgrade to 0.52?

Thanks in advance.

**Comment by zachowj** [comment]
If the host you have NR installed can access Home Assistant it doesn't really matter where each is installed.

> Finally, not really related, occasionally (frequently) when I deploy, one or two nodes say they cannot communicate with the server. Is there a solution to this, or will I have to downgrade to 0.52?

If you referencing the call-service node it is because your flows are getting to the call-service node before NR has connected to HA. Try changing your deploy method to "Modified Nodes".

**Reply by rjdza** [reply]
> If the host you have NR installed can access Home Assistant it doesn't really matter where each is installed.

Please excuse me if this is a stupid question then (I have googled and searched the docs).  How do I connect node-red to the plugin?

Things work when I connect to the url, but when I choose to use the HA plugin there are no further config options and nothing works :\

**Reply by zachowj** [reply]
What is the "plugin"? The Node-RED companion custom component? If so, you don't connect to it. You add it to HA using the integration menu.

https://github.com/zachowj/hass-node-red#installation -- step 8

**Reply by rjdza** [reply]
> If the host you have NR installed can access Home Assistant it doesn't really matter where each is installed.

I have the Home Assistant add-on installed under Home Assistant, and the installation documentation says all configuration is done in node red, but when I add a HA server in node-red and  tick the "I Use The Home Assistant Add-on" there are no further configuration options - where do I tell node-red how to connect to the add-on?

(Thanks for the help, btw.)

**Reply by zachowj** [reply]
If you're running NR on the host machine you are not using the Node-RED addon https://github.com/hassio-addons/addon-node-red

**Reply by rjdza** [reply]
> If you're running NR on the host machine you are not using the Node-RED addon https://github.com/hassio-addons/addon-node-red

As I mentioned in my initial post, I did install the Node-RED addon  using HACS, and then added it as an integration, so Home Assistant is definitely running it..

As to whether I'm running it on the host, that's a complex question, also mentioned in my initial post - I'm running Node-RED on the host, but I'm running Home Assistant in a docker instance, which means it's the same physical machine, but potentially a different machine since it's running in a separate namespace (basically a virtual machine).

How am I supposed to run the companion and integrate Node-RED?

**Comment by zachowj** [comment]
What you installed from HACS is not an "add-on" it's a custom component. Add-ons are docker containers built to work closely with the Home Assistant supervisor.
